### ISSUE TITLE
Iterative phase estimation not generating the correct circuit

### ISSUE THREAD

**Issue by shifubear** [Qiskit/qiskit-terra#9280]
### Environment

- **Qiskit Terra version**: 0.22.3
- **Python version**: 3.10.8
- **Operating system**: macOS Ventura 13.0.1

### What is happening?

The [iterative phase estimation class](https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/algorithms/phase_estimators/ipe.py) is not generating the correct circuit, causing the algorithm to behave incorrectly. The fix is very simple too, the controlled unitary is just using the incorrect qubit as the control. 

### How can we reproduce the issue?

Consider the unitary $T \otimes T$. This has the following (eigenvector, eigenvalue) pairs 
1. $(\left|00\right>$, $1)$
2. $(\left|01\right>$, $e^{i \pi/4})$
3. $(\left|10\right>$, $e^{i \pi/4})$
4. $(\left|11\right>$, $i)$ 

The following code runs QPE and IQPE using the fourth eigenvector, eigenvalue pair. 

```
from qiskit.providers.aer import AerSimulator
from qiskit.utils import QuantumInstance
from qiskit.opflow import X, T
from qiskit.algorithms.phase_estimators import (
    PhaseEstimation,
    IterativePhaseEstimation,
)

unitary_circuit = (T ^ T).to_circuit()
state_preparation = X^X  # Prepare the eigenstate |11> 

# Iterative phase estimation 
iqpe = IterativePhaseEstimation(
  num_iterations=3,
  quantum_instance=QuantumInstance(AerSimulator(method='statevector'), shots=1000)
)
iqperesult = iqpe.estimate(unitary_circuit, state_preparation=state_preparation.to_circuit())
print(f'IQPE result: {iqperesult.phase}')

# Phase estimation 
qpe = PhaseEstimation(
  num_evaluation_qubits=3,
  quantum_instance=QuantumInstance(AerSimulator(method='statevector'), shots=1000)
)
qperesult = qpe.estimate(unitary_circuit, state_preparation=state_preparation.to_circuit())
print(f'QPE result: {qperesult.phase}')
```

The above should generate the output 
```
IQPE result: 0.125
QPE result: 0.25

```

### What should happen?

Both algorithms should predict the same phase of 0.25, since $e^{2 \pi  i * 0.25} = i$. However, as the output above will show, IQPE outputs 0.125. A similar mistake can be observed for the second case. 

### Any suggestions?

I was able to fix this very easily by changing line 122 of the file https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/algorithms/phase_estimators/ipe.py: 

Before: 
```
qc = qc.compose(unitary_power, list(range(1, unitary.num_qubits + 1)) + [0])
```

After: 
```
qc = qc.compose(unitary_power, [unitary.num_qubits] + list(range(0, unitary.num_qubits)))
```

Reason for change: 
The control qubit should be the phase_register which is the final qubit, not the first qubit. 

Personal note: 
If possible I'd like to make the PR for this change, as it'll be my first code contribution to Qiskit.

**Comment by Cryoris** [comment]
@shifubear thanks for reporting this! It would be great if you want to fix this as a first contribution, see e.g. here for some guidance if you want: https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md. Updating this would also require adding a test to ensure that the behavior is now fixed, ideally in https://github.com/Qiskit/qiskit-terra/blob/main/test/python/algorithms/test_phase_estimator.py.

**Comment by woodsp-ibm** [comment]
This was fixed by #9887 but was not closed at the time. So closing it now.
